### PR TITLE
Split out remaining KO components

### PR DESCRIPTION
**Pull request by orangejenny**
## Technical Summary
Same idea as https://github.com/dimagi/commcare-hq/pull/35187, for the rest of our knockout components.

### Safety story
There's some risk that I missed something. This isn't a high-risk PR, but it's probably a medium-risk one.

The impact would be a widget not getting instantiated, which would mean it would be missing from the page, although it should break the other js on the page. Those kind of bugs can be quite visible.

I did this by grepping for all usages of these widgets (`<inline-edit`, `<pagination`, `<search-box`, `<select-toggle`) plus the `inline_edit_trans` template tag and `SelectToggle` crispy widget, and then grepping for `hqwebapp/js/components.ko` to find everywhere that module is used as either a script tag or a js dependency, and cross-referencing those two lists.

I'm going to do some targeted testing on staging, primarily to convince myself that the places where the js dependency was *missing* are still working properly and don't indicate I'm misunderstanding something.

When reviewing, it's worth knowing that non-requirejs page (mostly app manager and reports) don't *need* to have the components added to javascript modules. Modules that are only used on non-requirejs pages don't need to specify their dependencies at all, but some of them do, and in those cases I updated the dependency lists.

On non-requirejs pages, all that matters is that the relevant script tag is included, which for these components it always is, since they're in [core_libraries](https://github.com/dimagi/commcare-hq/blob/b38e1c874a0f1b1645ee3da9dc30a4fef24c9660/corehq/apps/hqwebapp/templates/hqwebapp/includes/core_libraries.html#L111-L115).

### Automated test coverage

I don't think there are any tests that any of these components get properly instantiated.

### QA Plan

Not requesting QA. See safety story.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Comment by orangejenny** [comment]
Not fixing lint errors.

File: corehq/apps/hqwebapp/static/hqwebapp/js/components/inline_edit.js
Comment (mjriley) [inline corehq/apps/hqwebapp/static/hqwebapp/js/components/inline_edit.js:156]: I notice this block in each component, as it used to be in _components.ko.js_. Do you think it makes sense to create a common registration function? Specifically, the `$el.data('apply-bindings')` logic isn't that obvious, so I worry about the duplication.

**Review by mjriley — COMMENTED** [review]
Thanks for doing this, I love these changes! That said, non-blocking concern over the duplication in widget registration. Stopping short of approval over a question of whether we need a bridge for in-flight code. Is it possible that someone has in-flight code using components.ko.js and won't see a merge conflict with these changes, but will introduce errors? Should we leave a placeholder components.ko.js that imports all these individual components as a bridge?

**Comment by orangejenny** [comment]
Noting that I tested all pages touched in this PR on staging to verify their expected widgets are shown.

File: corehq/apps/hqwebapp/static/hqwebapp/js/components/inline_edit.js
Comment (orangejenny) [reply corehq/apps/hqwebapp/static/hqwebapp/js/components/inline_edit.js:156]: yeah, I don't feel great about this duplication either, I'll take a look at cleaning it up.

File: corehq/apps/hqwebapp/static/hqwebapp/js/components/inline_edit.js
Comment (orangejenny) [reply corehq/apps/hqwebapp/static/hqwebapp/js/components/inline_edit.js:151]: Point taken, although I'm going to ignore it. I have a vague opinion we should move away from both underscore and jquery in favor of Array functions, but we don't have official guidance around this. It'd be a good discussion topic.

File: corehq/apps/hqwebapp/static/hqwebapp/js/components/inline_edit.js
Comment (orangejenny) [reply corehq/apps/hqwebapp/static/hqwebapp/js/components/inline_edit.js]: Cleaned up in 61776dbfb0d

File: corehq/apps/hqwebapp/static/hqwebapp/js/components/inline_edit.js
Comment (orangejenny) [reply corehq/apps/hqwebapp/static/hqwebapp/js/components/inline_edit.js]: Ended up addressing this in 61776dbfb0d because it did seem silly to pull in an underscore dependency *just* for `_.each`

**Comment by orangejenny** [comment]
@mjriley The question about bridging code is a good point. I'm inclined to just grep for `hqwebapp/js/components.ko` shortly before deploy to verify nothing new has been introduced. I don't think we have *that* much front end activity in the code base. Does that seem too sketchy?

File: corehq/apps/registry/static/registry/js/registry_edit.js
Comment (gherceg) [inline corehq/apps/registry/static/registry/js/registry_edit.js:17]: I don't know JS and am curious why these need to be included in here when components.ko wasn't included before?

**Comment by mjriley** [comment]
> I'm inclined to just grep for `hqwebapp/js/components.ko` shortly before deploy to verify nothing new has been introduced. I don't think we have _that_ much front end activity in the code base. Does that seem too sketchy?

I think I'll defer to your judgment here? If we were to import them all back into `components.ko.js`, probably with a note that importing from that was deprecated, I guess we run the 'risk' of people used to the old version never changing their code and having no real reason to do so. Do we have local hosters that have made their own changes? I don't know how realistic that concern is. And hopefully current in-development branches check their changes on staging after this is merged?

**Review by mjriley — APPROVED** [review]
Changes look great, thanks

**Comment by orangejenny** [comment]
@mjriley I'm not worried about localhosters having made code changes, just internal PRs. I'm comfortable with grepping for changes before deploy, I'll do that.

**Review by millerdev — APPROVED** [review]
Reviewed last two commits only.

File: corehq/apps/dashboard/static/dashboard/js/dashboard.js
Comment (orangejenny) [reply corehq/apps/dashboard/static/dashboard/js/dashboard.js:7]: I'm pretty confident. The grepping described in the safety story should have found all instances of the widgets, and that list is what I used to add/update dependencies, not the comments.

**Pull request merged**
